java-client/weather-server-example/src/main/docker/app.d/weather_server.py

```diff
@@ -0,0 +1,194 @@
+
+# Now make a table containing the last values of all the relevant intervals for each city.
+LastByCityState = combinedStats.lastBy("State", "City") \
+    .dropColumns("bin30M", "bin1Hr") \
+    .naturalJoin(CurrentData.lastBy("State", "City"), "State,City") \
+    .moveColumnsUp("Timestamp", "State", "City", "Temp", "Humidity")
+
+
+# Borrowed from https://stackoverflow.com/questions/27928/calculate-distance-between-two-latitude-longitude-points-haversine-formula
+def distance(lat1, lon1, lat2, lon2):
+    p = pi/180
+    a = 0.5 - cos((lat2-lat1)*p)/2 + cos(lat1*p) * cos(lat2*p) * (1-cos((lon2-lon1)*p))/2
+    return 12742 * asin(sqrt(a)) #2*R*asin...
+
+# Locate the Lat/Lon of a particular city state
+def geoLocate(cityName) -> Location:
+    if len(cityName) <= 0:
+        raise ValueError("Place name must be present")
+
+    # First go figure out Lat/Lon so we can pass that to NOAA and locate a station
+    geoResp = requests.get("https://maps.googleapis.com/maps/api/geocode/json", params={'address': cityName, 'key': API_KEY})
+    if geoResp.status_code != 200:
+        raise ValueError(cityName + " is not a valid place -> " + geoJson['error_message'])
+    geoJson = geoResp.json()
+
+    # Process the response JSON and look for the City and state (typically locality and administrative_area_level_1)
+    localCity = localState =''
+    resultsEl = geoJson['results'][0]
+    if resultsEl is None:
+        raise ValueError("Cannot determine location of " + cityName + " no valid results")
+
+    comps = resultsEl['address_components']
+    if comps is None:
+        raise ValueError("Cannot determine location of " + cityName + " no components")
+
+    for val in comps:
+        if 'locality' in val['types']:
+            localCity = val['long_name']
+        if 'administrative_area_level_1' in val['types']:
+            localState = val['long_name']
+
+    if localCity is None or localState is None:
+        raise ValueError("Unable to determine city and state for " + cityName)
+
+    geom = resultsEl['geometry']
+    if geom is None:
+        raise ValueError("Unable to determine lat/lon for " + cityName)
+
+    loc = geom['location'];
+    if loc is None:
+        raise ValueError("Unable to determine lat/lon for " + cityName)
+
+    print("Located " + localState +", " + localCity + " at [" + str(loc['lat']) + ", " + str(loc['lng']) + "]")
+    return Location(city=localCity, state=localState, lat=loc['lat'], lon=loc['lng'])
+
+# We have to poke NOAA's APIs so that we can find 1) the "Point" 2) The OBservation station and 3) the Forecast station
+# Once we collect this data we can construct an appropriate API call to fetch current / forecasted weather data
+def discoverWeather(loc):
+    # First, convert the Lat/lon into a point
+    pointJson = requests.get("https://api.weather.gov/points/" + str(loc.lat) + "%2C" + str(loc.lon)).json()
+    #    print(pointJson)
```

Comment:
looks like errors aren't possible here any more? seems like if this could be removed, and for stationJson below too?